crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -273,23 +289,72 @@ impl DeltaTableInputEndpointInner {
         }
     }
 
+    /// Compute the subset of columns in the Delta table schema that occur in the SQL
+    /// table declaration.
+    fn relevant_columns(&self, table: &DeltaTable) -> Vec<String> {
+        // Column names in the SQL schema.
+        let sql_columns = self
+            .schema
+            .fields
+            .iter()
+            .map(|field| field.name.name())
+            .collect::<Vec<String>>();
+
+        if let Some(table_schema) = table.schema() {
+            let delta_columns = table_schema
+                .fields()
+                .map(|f| f.name().clone())
+                .collect::<Vec<String>>();
+
+            // We need to be careful in checking whether a Delta column name occurs in
+            // sql_columns.  Delta doesn't seem to include case sensitivity information any
+            // any form in its schema.  So we conservatively check whether column name
+            // occurs in the SQL tables as is or whether a lowercase column name occurs
+            // in the set of SQL columns converted to lowercase.
+
+            let sql_columns = sql_columns.iter().cloned().collect::<BTreeSet<_>>();
+            let sql_columns_lowercase = sql_columns
+                .iter()
+                .map(|c| c.to_lowercase())
+                .collect::<BTreeSet<_>>();
+
+            delta_columns
+                .into_iter()
+                .filter(|c| {
+                    sql_columns.contains(c) || sql_columns_lowercase.contains(&c.to_lowercase())
+                })
+                .collect::<Vec<_>>()
+        } else {
+            sql_columns
+        }
+    }
+
     /// Load the entire table snapshot as a single "select * where <filter>" query.
     async fn read_unordered_snapshot(
         &self,
+        table: &DeltaTable,
         input_stream: &mut dyn ArrowStream,
         receiver: &mut Receiver<PipelineState>,
     ) {
-        // Execute the snapshot query; push snapshot data to the circuit.
-        info!(
-            "delta_table {}: reading initial snapshot",
-            &self.endpoint_name,
-        );
+        let columns = self.relevant_columns(table);
 
-        let mut snapshot_query = "select * from snapshot".to_string();
+        let column_names = columns
```

Comment:
good idea, I had to fix this bug in the SQL compiler too.
you should have a test where a column name is a reserved keyword

Reply:
but usually this also preserves case

Reply:
I tried different combinations of cases. This seems to work with datafusion SQL.